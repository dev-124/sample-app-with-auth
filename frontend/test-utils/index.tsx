// based on testing-library doc example

import { render, RenderOptions, RenderResult } from '@testing-library/react'
import React from 'react'

import { LocaleContextWrapper } from '../src/locales/locale-context'

const AllProviders: React.ComponentType = ({
  children
}: SamePropTypeOnly<React.ReactNode>) => {
  return (
    <LocaleContextWrapper pageContext={{ locale: 'en' }}>{children}</LocaleContextWrapper>
  )
}
const customRender = (
  ui: JSX.Element,
  options?:
    | Pick<
        RenderOptions<typeof import('@testing-library/dom/types/queries')>,
        'container' | 'baseElement' | 'hydrate' | 'wrapper'
      >
    | undefined
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
