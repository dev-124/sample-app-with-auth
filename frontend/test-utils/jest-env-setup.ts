// This file gets run automatically by Jest before every test
// therefore you donâ€™t need to add the imports to every single test file.
import '@testing-library/jest-dom/extend-expect'

// attach all the DOM Testing Library methods directly onto puppeteer's ElementHandle
// import 'pptr-testing-library/extend'
