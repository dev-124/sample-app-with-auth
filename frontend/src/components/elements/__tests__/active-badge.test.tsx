import React from 'react'

import { ActiveBadge } from '..'
import { render } from '../../../../test-utils'

describe('ActiveBadge', () => {
  it('renders a green true text badge with green background', () => {
    const { container, getByText } = render(<ActiveBadge isActive='true' />)

    const trueText = getByText('true')
    expect(trueText).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
  it('renders a false text badge with red background', () => {
    const { container, getByText } = render(<ActiveBadge isActive='false' />)

    const falseText = getByText('false')
    expect(falseText).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})
