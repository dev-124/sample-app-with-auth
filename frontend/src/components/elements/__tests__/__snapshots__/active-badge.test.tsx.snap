// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveBadge renders a false text badge with red background 1`] = `
<div>
  <div
    class="global-wrapper"
  >
    <span
      class="relative inline-block align-middle text-center px-4 py-2 leading-5"
    >
      <span
        aria-hidden="true"
        class="absolute inset-0 bg-error text-red-800 opacity-50 rounded-sm"
      />
      <span
        class="relative text-sm font-semibold "
      >
         
        false
      </span>
    </span>
  </div>
</div>
`;

exports[`ActiveBadge renders a green true text badge with green background 1`] = `
<div>
  <div
    class="global-wrapper"
  >
    <span
      class="relative inline-block align-middle text-center px-4 py-2 leading-5"
    >
      <span
        aria-hidden="true"
        class="absolute inset-0 bg-success text-green-800 opacity-50 rounded-sm"
      />
      <span
        class="relative text-sm font-semibold "
      >
         
        true
      </span>
    </span>
  </div>
</div>
`;
